src/modules/scraper/service.ts

```diff
@@ -28,138 +28,104 @@ export class ScraperService {
-          await this.publishMerkleDistributorBlocks(latestBlocks);
+        const blockNumber = await this.providers.getProvider(chainId).getBlockNumber();
+        this.logger.log(`latest block chainId: ${chainId} ${blockNumber}`);
+        const range = await this.determineBlockRange(
+          chainId,
+          blockNumber,
+          this.appConfig.values.web3.spokePoolContracts[chainId].startBlockNumber,
+          this.processedBlockRepository,
+        );
+        if (range) {
+          const queueMsg = { chainId, ...range };
+          await this.scraperQueuesService.publishMessage<BlocksEventsQueueMessage>(ScraperQueue.BlocksEvents, queueMsg);
+          // publish the block range to be processed with a delay of 60 seconds
+          await this.scraperQueuesService.publishMessage<BlocksEventsQueueMessage>(
+            ScraperQueue.BlocksEvents,
+            queueMsg,
+            { delay: 1000 * 60 },
+          );
+          this.logger.log(`publish chainId: ${chainId} from: ${range.from} to: ${range.to}`);
         }
       } catch (error) {
         this.logger.error(error);
       }
-      await wait(20);
-    }
-  }
-
-  public async publishBlocks(latestBlocks: Record<string, number>) {
-    this.logger.log(JSON.stringify(latestBlocks));
-    const blockRanges = await this.determineBlockRanges(
-      latestBlocks,
-      this.appConfig.values.web3.spokePoolContracts,
-      this.processedBlockRepository,
-    );
-
-    for (const chainId of Object.keys(blockRanges)) {
-      const { from, to } = blockRanges[chainId];
-      await this.scraperQueuesService.publishMessage<BlocksEventsQueueMessage>(ScraperQueue.BlocksEvents, {
-        chainId: parseInt(chainId),
-        from,
-        to,
-      });
+      await wait(secondsInterval);
     }
   }
 
-  public async publishMerkleDistributorBlocks(latestBlocks: Record<string, number>) {
-    const chainId = this.appConfig.values.web3.merkleDistributor.chainId;
-    const configStartBlockNumber = this.appConfig.values.web3.merkleDistributor.startBlockNumber;
-    const latestBlockNumber = latestBlocks[chainId];
-
-    const blockRange = await this.determineBlockRange(
-      chainId,
-      latestBlockNumber,
-      configStartBlockNumber,
-      this.merkleDistributorProcessedBlockRepository,
-    );
-
-    if (!blockRange) {
-      return;
+  public async publishMerkleDistributorBlocks(interval: number) {
+    while (true) {
+      try {
+        const chainId = this.appConfig.values.web3.merkleDistributor.chainId;
+        const blockNumber = await this.providers.getProvider(chainId).getBlockNumber();
+        const configStartBlockNumber = this.appConfig.values.web3.merkleDistributor.startBlockNumber;
+        const range = await this.determineBlockRange(
+          chainId,
+          blockNumber,
+          configStartBlockNumber,
+          this.merkleDistributorProcessedBlockRepository,
+          true,
+        );
+
+        if (range) {
```

Comment:
Nit: should we explicitly test if the range is undefined/null or collapse it into a boolean with `!!range`?

Reply:
Did it [here](https://github.com/across-protocol/scraper-api/pull/223/commits/4cd696bf06bf2c6e99603c713b8633a2e85be1a1)